src/cmds/scripts/win_postinstall.py

```diff
@@ -0,0 +1,632 @@
+import sys
+import getopt
+import os
+import subprocess
+import socket
+from tempfile import mkstemp
+from shutil import move
+import time
+
+install_path = ''
+pbs_path = ''
+
+
+def replace(file_path, pattern, subst):
+    fh, abs_path = mkstemp()
+    with os.fdopen(fh, 'w') as new_file:
+        with open(file_path) as old_file:
+            for line in old_file:
+                new_file.write(line.replace(pattern, subst))
+    # Remove original file
+    os.remove(file_path)
+    # Move new file
+    move(abs_path, file_path)
+
+
+def ValidateCredentials(username, password):
+    if (username != '' and password != ''):
+        with open(os.devnull, 'w') as fp:
+            pbs_account_binary = os.path.join(
+                pbs_path, "exec", "bin", "pbs_account.exe")
+            theproc = subprocess.Popen(
+                [pbs_account_binary,
+                    "-c", "-s", "-a", username, "-p", password],
+                stdout=fp,
+                stderr=fp)
+            theproc.communicate()
+            if theproc.returncode > 0:
+                print "Invalid credentials"
+                sys.exit(3)
+            else:
+                print "Post install process started..."
+                print "User Validation Success"
+    else:
+        print "Usage error: \n post_install.py -u <username> -p <password>"
+        sys.exit(4)
+
+
+def CreatePbsConfFile():
+    pbsconffile = os.path.join(pbs_path, 'pbs.conf')
+    with open(pbsconffile, 'w') as fp:
+        fp.write("PBS_SERVER=" + socket.gethostname() + "\r\n")
+        fp.write("PBS_EXEC=" + pbs_path + "exec\r\n")
+        fp.write("PBS_HOME=" + pbs_path + "home\r\n")
+        fp.write("PBS_START_SERVER=1\r\n")
+        fp.write("PBS_START_COMM=1\r\n")
+        fp.write("PBS_START_MOM=1\r\n")
+        fp.write("PBS_START_SCHED=1\r\n")
+        fp.close()
+        print "pbs.conf file created"
+
+
+def RunPbsmkdirs():
+    os.system('pbs_mkdirs')
+    print 'Securing and recreating pbs_home directories'
+
+
+def InstallingVcredist():
+    print 'Installing Visual C++ Redist..'
+    redist_path = os.path.join(install_path, "vcredist_x86.exe")
+    install_redist = '"' + redist_path + '"' + ' /passive /norestart'
+    if (os.path.exists(redist_path)):
+        os.system(install_redist)
+        print 'Installation done..'
+    else:
+        print "Visual C++ Redist is not available"
+        sys.exit(5)
+
+
+def UpdateDSconfig():
+    postgresql_file_path = os.path.join(
+        pbs_path, "home", "datastore", "postgresql.conf")
+    if (os.path.exists(postgresql_file_path)):
+        # change port to 15007
+        replace(postgresql_file_path, "#port = 5432", "port = 15007")
+        # change listen address to *
+        replace(
```

Comment:
For simple line statements, I have added those statement as it is. 
For complex unclear statements, have taken necessary actions using two variables